Koha/Plugin/Fi/KohaSuomi/DI/PatronController.pm

```diff
@@ -553,26 +554,59 @@ sub validate_credentials {
     }
 
     my $dbh = C4::Context->dbh;
-    unless (C4::Auth::checkpw_internal($dbh, $userid, $password)) {
-        return $c->render(
-            status => 401, 
-            openapi => { error => "Login failed." }
-        );
-    }
-
+    
     my $patron = Koha::Patrons->find({ userid => $userid });
-    $patron = Koha::Patrons->find({ cardnumber => $userid }) unless $patron;
-
-    if ($patron && $patron->lost) {
-        return $c->render( 
-            status => 403, 
-            openapi => { 
-                error => "Patron's card has been marked as 'lost'. Access forbidden." 
+    
+    #Try login if account is not locked
+    if ($patron and !$patron->account_locked) {
+        
+        unless (C4::Auth::checkpw_internal($dbh, $userid, $password)) {
+        
+            my $patron = Koha::Patrons->find({ userid => $userid });
+            if ($patron) {
+                $patron->update({ login_attempts => $patron->login_attempts + 1 });   
             }
-        );
-    }
 
-    return $c->render(status => 200, openapi => $patron->to_api);
+            return $c->render(
+                status => 401, 
+                openapi => { error => "Login failed." }
+            );
+        } 
+        
+        my $patron = Koha::Patrons->find({ userid => $userid });
+        $patron = Koha::Patrons->find({ cardnumber => $userid }) unless $patron;
+        
+        my @return = C4::Auth::checkpw_internal( $dbh, $userid, $password);
+        my $passwd_ok = 1 if $return[0] > 0; # 1 or 2
+        
+        if ($patron && !$patron->lost && $passwd_ok) {
+            my $lastseen = strftime "%Y-%m-%d %H:%M:%S", localtime;
+            $patron->update({ lastseen => $lastseen });
+            $patron->update({ login_attempts => 0 });
+            $patron->store;
+        }
+
+        if ($patron && $patron->lost) {
```

Comment:
Check this before updating lastseen and login_attempts. Then you can set lastseen and login_attempts unconditionally.